pywinauto/controls/win_base_wrapper.py

```diff
@@ -258,16 +260,22 @@ def drag_mouse_input(self,
 
         self.press_mouse_input(button, press_coords, pressed, absolute=absolute)
         time.sleep(Timings.before_drag_wait)
-        for i in range(5):
-            self.move_mouse_input((press_coords[0] + i, press_coords[1]), pressed=pressed, absolute=absolute) # "left"
-            time.sleep(Timings.drag_n_drop_move_mouse_wait)
-        self.move_mouse_input(release_coords, pressed=pressed, absolute=absolute) # "left"
+
+        if duration < 0.5:
+            for i in range(5):
+                self.move_mouse_input((press_coords[0] + i, press_coords[1]), pressed=pressed, absolute=absolute)
+                time.sleep(Timings.drag_n_drop_move_mouse_wait)
+
+        self.move_mouse_input(release_coords, pressed=pressed, absolute=absolute, duration=duration)
```

Comment:
This `duration` is not a duration of `drag_mouse_input`. The whole duration could be much longer. Looks not user-friendly.

Reply:
If duration > 0.5 than the whole duration could be 0.8 seconds longer.
```
for i in range(5) = 0.5 sec
before_drag_wait = 0.2 sec
before_drop_wait = after_drag_n_drop_wait = 0.1 sec
```
else the whole duration could be 0.3 seconds longer (not much).

Maybe if the duration is less than 0.5, the duration value must be reset to zero?